pkg/datapath/rlimit/rlimit.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018-2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rlimit
+
+import (
+	"math"
+
+	"golang.org/x/sys/unix"
+)
+
+// Configure configures the memory resource limits for the process to allow
+// BPF syscall interactions.
+func Configure() error {
```

Comment:
Do we want to move this into `pkg/datapath/linux` instead? It would be great if we could compile the entire agent with the fake datapath on non-Linux in the future.

Reply:
Good suggestion, will see how much it interferes with test includes (this was the main reason I didn't out it directly in datapath/linux already)

Reply:
@tgraf I moved it into a `sync.Once()` under `pkg/datapath/linux/datapath:NewDatapath()`, with the actual rlimit implementation in a separate file under `pkg/bpf`.

Reply:
Eh, didn't work with unprivileged tests. I've moved it back to `daemon.initMaps()`.